tests/transformers/test_transformers_model_export.py

```diff
@@ -3866,3 +3877,74 @@ def test_model_config_is_not_mutated_after_prediction(text2text_generation_pipel
     if validate_max_new_tokens:
         assert pyfunc_model.model_config["max_new_tokens"] == 500
         assert len(prediction_output[0].split(" ")) <= 5
+
+
+@pytest.mark.skipif(
+    Version(transformers.__version__) < Version("4.34.0"), reason="Feature does not exist"
+)
+def test_text_generation_task_chat_predict(text_generation_pipeline, model_path):
+    mlflow.transformers.save_model(
+        transformers_model=text_generation_pipeline,
+        path=model_path,
+        task="llm/v1/chat",
+    )
+
+    pyfunc_loaded = mlflow.pyfunc.load_model(model_path)
+
+    inference = pyfunc_loaded.predict(
+        {
+            "messages": [
+                {"role": "system", "content": "Hello, how can I help you today?"},
+                {"role": "user", "content": "How to learn Python in 3 weeks?"},
+            ],
+            "max_tokens": 10,
+        }
+    )
+
+    assert inference[0]["choices"][0]["message"]["role"] == "assistant"
+    assert (
+        inference[0]["choices"][0]["finish_reason"] == "length"
+        and inference[0]["usage"]["completion_tokens"] == 10
+    ) or (
+        inference[0]["choices"][0]["finish_reason"] == "stop"
+        and inference[0]["usage"]["completion_tokens"] < 10
+    )
+
+
+@pytest.mark.skipif(
+    Version(transformers.__version__) < Version("4.34.0"), reason="Feature does not exist"
+)
+def test_text_generation_task_chat_serve(text_generation_pipeline):
+    data = {
+        "messages": [
+            {"role": "user", "content": "How to learn Python in 3 weeks?"},
+        ],
+        "max_tokens": 10,
+    }
+
+    with mlflow.start_run():
+        model_info = mlflow.transformers.log_model(
+            transformers_model=text_generation_pipeline,
+            artifact_path="model",
+            task="llm/v1/chat",
+        )
+
+    inference_payload = json.dumps({"inputs": data})
```

Comment:
does it not work just with `data`? i would have thought that having the `messages` key prevents the user from having to put this all behind `inputs`

Reply:
Yeah it would work, I'll remove the `PredictionsResponse` line then, just copied the test from somewhere else.

Reply:
Sounds good! Yeah I think we can just do a `json.loads` on the response content and check the fields that way (at least I imagine that's what most users would be doing with this feature)